Add sample GE WPF Client and Missing Security Bits for GE/SF integration

TaviTruman opened the pull request:
Adding WPF Client samples that demonstrate GE APIs; adding Security bits for GE/SF

TaviTruman committed:
Merge pull request #1 from Microsoft/master

Bring in line with Master

TaviTruman committed:
changes to reflect local azure service fabric configuration

TaviTruman committed:
Local Service Fabric changes

TaviTruman committed:
Merge branch 'master' of https://github.com/Microsoft/GraphEngine

TaviTruman committed:
Local updates

TaviTruman committed:
Merge branch 'master' of https://github.com/Microsoft/GraphEngine

TaviTruman committed:
Add new Service Fabric Examples

TaviTruman committed:
Merge branch 'master' of https://github.com/Microsoft/GraphEngine

TaviTruman committed:
Add sample Hello World WPF Client and Missing Security Bits from Service Fabric

yatli commented on src/Modules/GraphEngine.ServiceFabric/Trinity.ServiceFabric.GraphEngineService/PackageRoot/Data/trinity.xml:
```diff
@@ -4,7 +4,8 @@
     <Trinity.LIKQ Timeout="10000"/>
     <!-- In this demo application we use Azure blob storage as our persistent storage. -->
-    <Trinity.Azure.Storage ConnectionString="your_connection_string" ContainerName="graphengine"/>
+    <Trinity.Azure.Storage ConnectionString="DefaultEndpointsProtocol=https;AccountName=ikwstorage;AccountKey=tYAAfASvWRU7AjRAHoAwBd08WnUW3uimR3KIdOVCOC55cfSrCJNx7tEbz3jmbEwWWN6fRbYI0U6MY3UPXhRZ8A==;EndpointSuffix=core.windows.net" ContainerName="graphenginestoragespace 
```
please remove your key

TaviTruman replied:
Done!

yatli commented on src/Modules/GraphEngine.ServiceFabric/Trinity.ServiceFabric.GraphEngineService/PackageRoot/Data/trinity_large_memory.xml:
```diff
@@ -4,7 +4,8 @@
     <Trinity.LIKQ Timeout="10000"/>
     <!-- In this demo application we use Azure blob storage as our persistent storage. -->
-    <Trinity.Azure.Storage ConnectionString="your_connection_string" ContainerName="graphengine"/>
+    <Trinity.Azure.Storage ConnectionString="DefaultEndpointsProtocol=https;AccountName=ikwstorage;AccountKey=kcdkVkeKMjlu+u8mRxnRLOcApQRnFbrXbVQE3j++RPvRdC45FkRThgOnkyaBcn7l3p3nCpICv5D0gzGE9XjAow==;EndpointSuffix=core.windows.net" ContainerName="graphenginestoragespace 
```
please remove your key

TaviTruman replied:
@yatli Yikes - forgot to put the default value back.

TaviTruman replied:
@yatli I have also upgraded the SF SDK and runtime support to 3.3.654; only a few simple code changes required.

yatli commented on src/Modules/GraphEngine.ServiceFabric/Trinity.ServiceFabric.SampleApplication/ApplicationPackageRoot/ApplicationManifest.xml:
```diff
@@ -2,9 +2,9 @@
   <Parameters>
     <Parameter Name="Trinity.ServiceFabric.GraphEngineRemotingClient_InstanceCount" DefaultValue="-1" />
-    <Parameter Name="Trinity.ServiceFabric.GraphEngineService_MinReplicaSetSize" DefaultValue="3" />
+    <Parameter Name="Trinity.ServiceFabric.GraphEngineService_MinReplicaSetSize" DefaultValue="2" />
```
please elaborate why to make this change.

TaviTruman replied:
@yatli I have done quite a bit of testing trying to understand the policy-behavior of the GE when running in the Service Fabric Cluster Configuration. I'm now running on Azure Stack with Service Fabric configured; I changed these setting because they consistently work when the cluster is in crisis under recovery and or update; with the default configuration the length of time required to restart the services has proven to be problematic. I have not seen any problems with scaling up. I now have 12 GE/SF App Services running serving up some 250,000,000 Hyper-Graph RDF nodes and it is awesome; a 5x(Apps)x5x5x10 configuration works well for my services. Another reason why I changed the cluster configuration was due to the length of time it was taking for services to come ready and if the Sample GE App would ever come back to ready-state.

Here's my CPU and Memory blueprint 

![image](https://user-images.githubusercontent.com/5692812/56984168-4451d680-6b3a-11e9-86bb-63fb8f994466.png)

![image](https://user-images.githubusercontent.com/5692812/56983915-b118a100-6b39-11e9-882c-4cdcdc2b7e0e.png)

Here's the GE/SF apps running very well in the 5x1x2x2x3 configuration. I take cut power or do oderly shutdonw and upgrade (replace) the services and it always comes back in under 2 minutes.

![image](https://user-images.githubusercontent.com/5692812/56984056-02289500-6b3a-11e9-93d4-beb1737ab43b.png)

I am still making a few changes in how the GE App service is written as I am sure that has something to do with it.

TaviTruman committed:
Adding WPF Sample calling GE APIs and GE SF App service

TaviTruman committed:
Restore default value in trinity.xml file in the GraphEngineService project; upgrade to Service Fabric 6.4 SDK.
Corrected breaking-change in SF Remoting API from Trinity.ServiceFabric.Remoting

TaviTruman committed:
Restored default value for Trinity.Azure.Storage ConnectionString

TaviTruman committed:
Restore default value for Trinity.Azure.Storage ConnectionString

TaviTruman committed:
Merge branch 'master' of https://github.com/InKnowWorks/GraphEngine

TaviTruman committed:
Restore default value for Trinity.Azure.Storage ConnectionString

TaviTruman committed:
Take changes coming from Merge

TaviTruman committed:
local changes from ikw master

TaviTruman committed:
Merge branch 'master' of https://github.com/InKnowWorks/GraphEngine

yatli commented on src/Modules/GraphEngine.DynamicCluster/Trinity.Azure.Storage.Test/Trinity.Azure.Storage.Test.csproj:
```diff
@@ -7,7 +7,7 @@
 
   <ItemGroup>
-    <PackageReference Include="GraphEngine.Core" Version="$(DOTNET_PACKAGE_VERSION)" />
+    <PackageReference Include="GraphEngine.Core" Version="2.0.9608" />
```
please rollback the changes specifying a fixed version. $(DOTNET_PACKAGE_VERSION) is introduced to update the version automatically.

TaviTruman replied:
@yatli Yep - I already caught it; local compilations and builds work against the build process as VS is making these changes to the project configuration files. Nuget Package Manager is overriding. I'm on it now.

TaviTruman replied:
Changed applied

TaviTruman committed:
Restore project settings removing literal values for GE versions

TaviTruman committed:
Restore project setting for build

TaviTruman committed:
Update SF Cluster Configuration for HelloWorldAPI

TaviTruman committed:
Correct SF Cluster configuration

TaviTruman committed:
Merge branch 'master' of https://github.com/microsoft/GraphEngine

TaviTruman committed:
Sync all projects with Microsoft GraphEngine Master Repro; removed new samples the GraphEngine Service Fabric projects

TaviTruman committed:
Bring supporting library up to date

TaviTruman committed:
Moving new samples around and create new documentation folders

TaviTruman committed:
Updated Service Fabric libraries and changed ITrinityOverRemotingService to use V2 Remoting Client

TaviTruman committed:
Set values back to orginial settings

TaviTruman committed:
Updated Service Fabric librries to the most recent version and release

TaviTruman committed:
Removed WPF and Service Fabric  Samples  from Graph Engine Service directory

TaviTruman committed:
Added new Sub-directries for new Samples

TaviTruman committed:
Renamed  new Samples Sub-directories

TaviTruman committed:
Added new Contacts Service Fabric App Solution and SF/GE Services

TaviTruman committed:
Added Graph Engine CommunicationProtocolModel and GE Property Graph Data Model for Contact Entity Management

TaviTruman committed:
Updated Friends Sample Add use to the Latest Build of the Graph Engine: Build 2.0.10065

TaviTruman committed:
Updated the Ping sample to the latest build and made minor code changes due to revisions in the API

TaviTruman committed:
Updated Samples apps to use recent build of Graph Engine; made small code corrections due to changed in GE Code Generation. Updated Graph Engine VS Entension.

TaviTruman committed:
Merge branch 'master' of https://github.com/microsoft/GraphEngine

TaviTruman committed:
Merge changes taken from master

TaviTruman committed:
Updated app to use .netcoreapp3.0

TaviTruman committed:
Reset Target

TaviTruman committed:
Updated sample GE apps to .NET Core 3.0 to test

TaviTruman committed:
Upgrade to new build of Graph Engine

yatli commented on .gitignore:
```diff
@@ -113,3 +113,4 @@ CMakeSettings.json
 Directory.Build.props
 .ionide
 launchSettings.json
+/out/build/x64-Debug
```
revert this please.

yatli commented on Folder.DotSettings:
```diff
@@ -0,0 +1,10 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
```
What's the purpose of this file?

TaviTruman replied:
This is part of ReSharper - I will delete it as it is not required.

yatli commented on out/build/x64-Debug/VSInheritEnvironments.txt:
```diff
@@ -0,0 +1 @@
+msvc_x64_x64
```
don't check-in out/build

TaviTruman replied:
got it

yatli commented on samples/Friends/Friends/App.config:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
please revert. samples updated.

yatli commented on samples/Friends/Friends/Friends.csproj:
```diff
@@ -2,15 +2,15 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
```
please revert.

yatli commented on samples/Friends/Friends/Friends.csproj:
```diff
@@ -2,15 +2,15 @@
 
   <ItemGroup>
-    <PackageReference Include="GraphEngine.Core" Version="2.0.0" />
+    <PackageReference Include="GraphEngine.Core" Version="2.0.10084" />
```
please revert. 10084 is a specific build and may not exist.

TaviTruman replied:
Done

yatli commented on samples/Friends/Friends/Program.cs:
```diff
@@ -15,8 +15,9 @@ namespace Friends
-        public unsafe static void Main(string[] args)
+        public static unsafe void Main(string[] args)
         {
+            TrinityConfig.CurrentRunningMode = RunningMode.Embedded;
```
Please revert.

TaviTruman replied:
Just altered code formatting - that line of code has always been there

yatli commented on samples/Friends/Friends/packages.config:
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
delete this.

TaviTruman replied:
Was added by VS 2019 - I've removed the statement.

yatli commented on samples/GraphGenerator/GraphGenerator/App.config:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
delete this.

TaviTruman replied:
deleted

yatli commented on samples/GraphGenerator/GraphGenerator/packages.config:
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
delete this.

TaviTruman replied:
deleted

yatli commented on samples/Ping/Ping/App.config:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
delete this.

TaviTruman replied:
done

yatli commented on samples/Ping/Ping/Ping.csproj:
```diff
@@ -2,15 +2,15 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
```
revert to 2.0

TaviTruman replied:
done

yatli commented on samples/Ping/Ping/packages.config:
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
delete this.

yatli commented on samples/SSSP/SSSP/App.config:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
delete this.

yatli commented on samples/SSSP/SSSP/SSSP.csproj:
```diff
@@ -10,7 +10,7 @@
 
   <ItemGroup>
-    <PackageReference Include="GraphEngine.Core" Version="2.0.0" />
+    <PackageReference Include="GraphEngine.Core" Version="2.0.10068" />
```
revert

yatli commented on samples/SSSP/SSSP/packages.config:
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
delete this.

yatli commented on src/GraphEngine.DataImporter/GraphEngine.DataImporter/CmdOptions.cs:
```diff
@@ -21,16 +21,16 @@ class CmdOptions
         [Option('o', "output", HelpText = "Specifies data import output directory for importing tasks, and specifies the output TSL file name for TSL generation tasks", Required = false)]
         public string Output { get; set; }
 
-        [Option('g', "generate_tsl", HelpText = "Generates TSL", MutuallyExclusiveSet = "Action")]
```
👍

yatli commented on src/GraphEngine.DataImporter/GraphEngine.DataImporter/GraphEngine.DataImporter.csproj:
```diff
@@ -7,9 +7,9 @@
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="CommandLineParser" Version="1.9.71" />
+    <PackageReference Include="CommandLineParser" Version="2.6.0" />
```
👍

yatli commented on src/GraphEngine.DataImporter/GraphEngine.DataImporter/JsonImporter.cs:
```diff
@@ -79,7 +80,7 @@ public ICell ImportEntity(string type, string line, long? parent_id)
             tree_imports.ForEach(_ => _());
 
-            if (cellid == null) cellid = Global.LocalStorage.NewRandomCellID();
+            if (cellid == null) cellid = CellIdFactory.NewCellId();
```
👍

yatli commented on src/GraphEngine.DataImporter/GraphEngine.DataImporter/Program.cs:
```diff
@@ -19,9 +19,32 @@ internal class Program
         {
             CmdOptions opts = new CmdOptions();
-            if (CommandLine.Parser.Default.ParseArguments(args, opts))
+
```
revert this. don't hard-code local settings in the program.

TaviTruman replied:
Yikes - I was testing and forgot to remove testing code.

yatli commented on src/Modules/GraphEngine.DynamicCluster/Trinity.DynamicCluster/Communication/DynamicClusterCommModule.cs:
```diff
@@ -178,15 +178,15 @@ public override void GetChunksHandler(GetChunksResponseWriter response)
         public override void AnnounceMasterHandler(StorageInformationReader request, ErrnoResponseWriter response)
         {
-            (m_memorycloud as DynamicMemoryCloud).m_cloudidx.SetMaster(request.partition, request.id);
+            (m_memorycloud as DynamicMemoryCloud)?.m_cloudidx.SetMaster(request.partition, request.id);
```
nit: should not silently return E_OK when m_memorycloud resolution fails.

TaviTruman replied:
Restored to the original statement.

yatli commented on src/Modules/GraphEngine.DynamicCluster/Trinity.DynamicCluster/Communication/DynamicClusterCommModule.cs:
```diff
@@ -178,15 +178,15 @@ public override void GetChunksHandler(GetChunksResponseWriter response)
             try
             {
-                (m_memorycloud as DynamicMemoryCloud).m_backupctl.RestoreCurrentPartition(request.task_id, request.version, EventArgs.Empty).Wait();
+                (m_memorycloud as DynamicMemoryCloud)?.m_backupctl.RestoreCurrentPartition(request.task_id, request.version, EventArgs.Empty).Wait();
```
same here.

TaviTruman replied:
Restored to the original statement.

TaviTruman replied:
Reverted

yatli commented on src/Modules/GraphEngine.DynamicCluster/Trinity.DynamicCluster/Communication/DynamicClusterCommModule.cs:
```diff
@@ -199,7 +199,7 @@ public override void PersistedSavePartitionHandler(BackupTaskInformationReader r
         {
             try
             {
-                (m_memorycloud as DynamicMemoryCloud).m_backupctl.BackupCurrentPartition(request.task_id, request.version, EventArgs.Empty).Wait();
```
same here.

TaviTruman replied:
Restored to the original statement.

TaviTruman replied:
Reverted

yatli commented on src/Modules/LIKQ/LIKQ.sln.DotSettings:
```diff
@@ -0,0 +1,2 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
```
why wpf in LIKQ solution?

TaviTruman replied:
Removed - I've WPF and Service Fabric samples been removed from the Module directory and placed in sub-directory under samples.

yatli commented on src/Trinity.TSL/Trinity.TSL.Parser/tb_semantc_check.cpp:
```diff
@@ -42,6 +42,8 @@ inline string get_key(NProtocolProperty* target)
     case PT_STRUCT_RESPONSE:
         return "NProtocolProperty::ResponseType";
         break;
+      case PT_UNDEFINED: break;
```
these cases are covered in the throw.
please revert.

TaviTruman replied:
statement reverted

yatli commented on src/Trinity.VSExtension/Trinity.TSL.ManagedWrapper/Trinity.TSL.ManagedWrapper.vcxproj:
```diff
@@ -15,6 +15,7 @@
     <RootNamespace>TrinityTSLParserManagedWrapper</RootNamespace>
     <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
     <ProjectName>Trinity.TSL.ManagedWrapper</ProjectName>
+    <WindowsTargetPlatformVersion>10.0.18956.0</WindowsTargetPlatformVersion>
```
please revert. I'm working on the compatibility issues. thanks.

TaviTruman replied:
Done.

yatli commented on src/Trinity.VSExtension/TrinityVSExtension/TrinityVSExtension.csproj:
```diff
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="12.0">
-  <Import Project="..\packages\Microsoft.VSSDK.BuildTools.15.8.3252\build\Microsoft.VSSDK.BuildTools.props" Condition="Exists('..\packages\Microsoft.VSSDK.BuildTools.15.8.3252\build\Microsoft.VSSDK.BuildTools.props')" />
```
please revert. working on compatibility issues.

TaviTruman replied:
Done

yatli commented on src/Trinity.VSExtension/TrinityVSExtension/packages.config:
```diff
@@ -1,4 +1,4 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.VSSDK.BuildTools" version="15.8.3252" targetFramework="net45" developmentDependency="true" />
```
please revert. working on compatibility issues.

TaviTruman replied:
Done

yatli reviewed (changes requested):
I've gone through the changes again. Thanks Tavi, and let's clean this up.

TaviTruman committed:
Revert "Updated sample GE apps to .NET Core 3.0 to test"

This reverts commit 5c9085bab1a50597d60147107c5a26aac254a75c.

TaviTruman committed:
Roll back changes to Trinity VS extension

TaviTruman committed:
Removed LINQ.sln.DotSettings file and update Friends project to reflect GraphEngine 2.0.0 package

TaviTruman committed:
Removed encoding statement from package file

TaviTruman committed:
Removed encoding statements

TaviTruman committed:
Upgraded to the latest version of CommandLineParser and adjusted the code to reflect API changes.

TaviTruman committed:
Reverted / Removed  /out/build/x64-Debug stateement

TaviTruman committed:
Revert code back to orignial state and cleaned/removed files added by VS

TaviTruman reviewed (commented):
Yatao I have cleaned up the codebase but I still see that the CI builds aren't clean yet. Can you take a look and let me know what still needs correction. I can't see the CI build log.

TaviTruman committed:
Added new Sample WPF and Console Apps demonstrating Server-side push to Client

TaviTruman committed:
Merge branch 'master' of https://github.com/microsoft/GraphEngine into ikw-graphengine

# Conflicts:
#	src/Trinity.VSExtension/TrinityVSExtension/app.config

TaviTruman committed:
Updates to master adding in new samples

TaviTruman committed:
Renamed solution and projects; clearned up TPL threads

TaviTruman committed:
convert project to netcoreapp3.0

TaviTruman committed:
Renamed directories

TaviTruman committed:
Updated Service Fabric Libraries

TaviTruman committed:
Merge branch 'master' into ikw-graphengine

TaviTruman committed:
Updated the Service Fabric runtime and support libraries

TaviTruman committed:
Upgrade all projects to .NETCore 3.1

TaviTruman committed:
Adding test for multi-module support

TaviTruman committed:
Adjusting Test Server GE Server configuration

TaviTruman committed:
Adjusting Test Server GE Server configuration in support of Graph Engine Service and Client side Modularity. Updated build process to create optionally named Nuget repository.

TaviTruman committed:
Merge branch 'ikw-graphengine' of https://github.com/InKnowWorks/GraphEngine into ikw-graphengine

# Conflicts:
#	samples/WPF Graph Engine Client/IKW.Trinity.Client.TestServer/Trinity.WPF.TestClient/MainWindow.xaml.cs

TaviTruman committed:
Revert "Merge branch 'ikw-graphengine' of https://github.com/InKnowWorks/GraphEngine into ikw-graphengine"

This reverts commit 53d872c39160ea1e02632b5d111d9e50d62c4225, reversing
changes made to bc66506e86fefb726cc4e5e47f4ce1e5eebdef15.

TaviTruman committed:
- Adding support for gRPC for Trinity SF support
- Made spelling and naming corrections
- All Service Fabric support libraries are now .netstanard2.0
- Implementing support for Azure AD Managed Identifites
- Improving support for Azure Storage Blob support with Graph Engine stored images
- Code Samples and Documented Guidance proper suppot for Graph Engine Module App model

TaviTruman committed:
Extending TrinityClient implementation to support custom changed to the UniformInt64Partition in the service fabric Application Manifest. The current implementation only works when the default values for LowKey and HighKey properties.

TaviTruman committed:
- Updated the csproj files for TrinityVSExtension.VS2017 and VS2019 projects to fix build errors "Visual Studio Extension build failing with Error VSSDK1077: Unable to locate the extensions directory. "Value cannot be null"
- Updating TrinityClient so that is works outside of the Service Fabric Cluster; the current implementation seems to work well through extensive testing; the current implementation only works with Service Fabric with UniformInt64Partition when values for LowKey="-9223372036854775808" HighKey="9223372036854775807".If you change these values to say 1 to 10 then the randomly generated integer passed to the ServicePartitionKey method fails.
- Extending the TrinityClient so that one can pass into the constructor the partitionkey.
- A new TrinityClient that support the client connecting to the SF Partition

TaviTruman committed:
Small changes to Trinity.Client making it possible to specifiy the service fabric service partition key
Added guardian clauses in TrinityClientModule vis Try/Finally statement
Replaced ServiceEnpoint with ServiceEndpointV2 in service manifest

TaviTruman committed:
Small tweaks to source code; commenting and variable name improvementns

TaviTruman committed:
Global update and change to Build scripts in preperation to moving to the latest version of net core 3.1 and net standard 2.1

TaviTruman committed:
Updating projects in prepertation of migrationg to .NET 5; updated various build scripts trying to find the right updates in order to make the CLI build work.

TaviTruman committed:
Included Service Initialization and Request event sources

TaviTruman committed:
Upgrade Service Fabric Libraries

TaviTruman committed:
Applied small code changes to DynamicCluster and to DynamicCluster Test due to breaking changes in the Azure Storage APIs

TaviTruman committed:
Updated to the latest Service Fabric release and added new Service Fabric project

TaviTruman committed:
Adjusting code so as to mirror orginal soruce

TaviTruman committed:
Updated project directory structure

TaviTruman committed:
Adjusted source code and project settings to address this build error: O:\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.TargetFrameworkInference.targets(112,5): error : The current .NET SDK does not support targeting .NET Core 2.0.  Either target .NET Core 1.1 or lower, or use a version of the .NET SDK that su

pports .NET Core 2.0. [G:\IKW-GraphEngine\src\Modules\GraphEngine.ServiceFabric\Trinity.ServiceFabric.GraphEngine.GraphDataService\Trinity.ServiceFabric.GraphEngine.GraphDataService.csproj] [G:\IKW-GraphEngine\build\BUILD_GraphEngine.ServiceFabric.vcxproj]
O:\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.TargetFrameworkInference.targets(112,5): error : The current .NET SDK does not support targeting .NET Core 2.0.  Either target .NET Core 1.1 or lower, or use a version of the .NET SDK that su
pports .NET Core 2.0. [G:\IKW-GraphEngine\src\Modules\GraphEngine.ServiceFabric\Trinity.ServiceFabric.GraphEngine.RemotingClient\Trinity.ServiceFabric.GraphEngine.RemotingClient.csproj] [G:\IKW-GraphEngine\build\BUILD_GraphEngine.ServiceFabric.vcxproj]

TaviTruman committed:
Remove global.json file as it breaks the Graph Engine Build process

TaviTruman committed:
made minor correction to RemotingClient service initialiation code.

TaviTruman committed:
Small adjustments to source

TaviTruman committed:
Merge branch 'master' of https://github.com/microsoft/GraphEngine

TaviTruman committed:
Merged update

TaviTruman committed:
Upgrade Service Fabric packages to .netcoreapp3.1

TaviTruman committed:
Moved Sqmple Service Fabric solution and project to the Samples folder

TaviTruman committed:
.NET Core updates

TaviTruman committed:
Re-design the server and client for Reactive Extensions programming and support

TaviTruman committed:
Updated stock samples and revised testing suite for GE/SF

TaviTruman committed:
Added Trinity.ServiceFabric.NativeClient.Remoting module in support for external GE Client to GE Server in SF Cluster.

TaviTruman committed:
Added new Service Fabric Trinity Native Remoting Client-service  in support for External connection requestion from external Graph Engine Client applications

TaviTruman committed:
Upgrade to latest version of Service Fabric SDK; New GE Native Communications Proxy for Externally connnected GE (native) Client
Updates to GE/SF Test Harness

TaviTruman committed:
Updated Azure Storage Blob API and update SF Test program

TaviTruman committed:
Updating to the lastest releases of libraries

TaviTruman committed:
Upgrade DataImporter to .net core 3.1

TaviTruman committed:
Update to latest .net core 3.1

TaviTruman committed:
Update to lastest Graph Engine build

TaviTruman committed:
Updates to a number of .NET CORE Libraries and update to the latest version of Azure Service Fabric; also, adding support for new Azure Storage. Adding support for more integrated TCP/IP support in the Service Fabric

TaviTruman committed:
Added solution file

TaviTruman committed:
Updated global.json to .net core sdk

TaviTruman committed:
added --framework switch on cmake

TaviTruman committed:
Update to new build of GE

dependabot-preview[bot] committed:
Upgrade to GitHub-native Dependabot

TaviTruman committed:
Upgraded build process to support .net5.0; also, added native setup and configuration for Graph Engine Cluster.

TaviTruman committed:
Upgraded to .net 5

TaviTruman committed:
removed connection string apikey

TaviTruman committed:
Merge pull request #2 from InKnowWorks/dependabot/add-v2-config-file

Upgrade to GitHub-native Dependabot
